test/execution/execute-afm.test.ts

```diff
@@ -1,137 +1,567 @@
+            expect(getNextLimit([3, 2], [8, 0], [8, 2])).toEqual([0, 2]);
+        });
+
+        it('should return correct values for total=[5,5]', () => {
+            expect(getNextOffset([5, 5], [0, 0], [5, 5])).toEqual([5, 0]);
+            expect(getNextLimit([5, 5], [5, 0], [5, 5])).toEqual([0, 5]);
+        });
+
+        it('should return correct values for total=[5,10]', () => {
+            expect(getNextOffset([5, 5], [0, 0], [5, 10])).toEqual([0, 5]);
+            expect(getNextLimit([5, 5], [0, 5], [5, 10])).toEqual([5, 5]);
+
+            expect(getNextOffset([5, 5], [0, 5], [5, 10])).toEqual([5, 0]);
+            expect(getNextLimit([5, 5], [5, 0], [5, 10])).toEqual([0, 5]);
+        });
+    });
+});
+
+describe('mergePage', () => {
+    describe('1 dimension', () => {
+        function getExecutionResult(offset: number, total: number): Execution.IExecutionResult {
+            return {
+                headerItems: [
+                    [
+                        [
+                            A1
+                        ]
+                    ]
+                ],
+                data: [
+                    offset + 1
+                ],
+                paging: {
+                    count: [1],
+                    offset: [offset],
+                    total: [total]
+                }
+            };
+        }
+
+        it('should return correct result after merge 2 executionResults', () => {
+            expect(mergePage(
+                getExecutionResult(0, 2),
+                getExecutionResult(1, 2)
+            )).toEqual({
+                headerItems: [
+                    [
+                        [
+                            A1
+                        ]
+                    ]
+                ],
+                data: [
+                    1,
+                    2
+                ],
+                paging: {
+                    count: [2],
+                    offset: [0],
+                    total: [2]
+                }
+            });
+        });
+
+        it('should return correct result after merge 3 executionResults', () => {
+            const prevExecutionResult = mergePage(
+                getExecutionResult(0, 3),
+                getExecutionResult(1, 3)
+            );
+
+            expect(mergePage(
+                prevExecutionResult,
+                getExecutionResult(2, 3)
+            )).toEqual({
+                headerItems: [
+                    [
+                        [
+                            A1
+                        ]
+                    ]
+                ],
+                data: [
+                    1,
+                    2,
+                    3
+                ],
+                paging: {
+                    count: [3],
+                    offset: [0],
+                    total: [3]
+                }
+            });
+        });
+    });
+
+    describe('2 dimensions', () => {
```

Comment:
this test is too long. could we move out all test fixtures into separate file?

Reply:
I think that the test will be more readable if inputs and expects will be in the same file. Moreover, we would have to use helper functions for creating measures and attributes from this test file in fixtures file.